code/game/objects/structures/crates_lockers/closets/bodybag.dm

```diff
@@ -148,3 +149,159 @@
+	foldedbag_path = /obj/item/bodybag/environmental/prisoner/
+	breakout_time = 4 MINUTES // because it's probably about as hard to get out of this as it is to get out of a straightjacket.
+	var/sinch_time = 10 SECONDS
+	var/sinched = FALSE
+
+/obj/structure/closet/body_bag/environmental/prisoner/update_icon()
+	. = ..()
+	if(sinched)
+		icon_state = initial(icon_state) + "_sinched"
+	else
+		icon_state = initial(icon_state)
+
+/obj/structure/closet/body_bag/environmental/prisoner/can_open(mob/living/user, force = FALSE)
+	if(force)
+		return TRUE
+	if(sinched)
+		to_chat(user, span_danger("The buckles on [src] are sinched down, preventing it from opening."))
+		return FALSE
+	. = ..()
+
+/obj/structure/closet/body_bag/environmental/prisoner/open(mob/living/user, force = FALSE)
+	if(!can_open(user, force))
+		return
+	if(opened)
+		return
+	sinched = FALSE
+	playsound(loc, open_sound, open_sound_volume, TRUE, -3)
+	opened = TRUE
+	if(!dense_when_open)
+		set_density(FALSE)
+	dump_contents()
+	update_appearance()
+	after_open(user, force)
+	return TRUE
+
+/obj/structure/closet/body_bag/environmental/prisoner/container_resist_act(mob/living/user)
+	/// copy-pasted with changes because flavor text as well as some other misc stuff
+	if(opened)
+		return
+	if(ismovable(loc))
+		user.changeNext_move(CLICK_CD_BREAKOUT)
+		user.last_special = world.time + CLICK_CD_BREAKOUT
+		var/atom/movable/location = loc
+		location.relay_container_resist_act(user, src)
+		return
+	if(!sinched)
+		open()
+		return
+
+	user.changeNext_move(CLICK_CD_BREAKOUT)
+	user.last_special = world.time + CLICK_CD_BREAKOUT
+	user.visible_message(span_warning("Someone in [src] begins to wriggle!"), \
+		span_notice("You start wriggling, attempting to loosen [src]'s buckles... (this will take about [DisplayTimeText(breakout_time)].)"), \
+		span_hear("You hear straining cloth from [src]."))
+	if(do_after(user,(breakout_time), target = src))
+		if(!user || user.stat != CONSCIOUS || user.loc != src || opened || !sinched )
+			return
+		//we check after a while whether there is a point of resisting anymore and whether the user is capable of resisting
+		user.visible_message(span_danger("[user] successfully broke out of [src]!"),
+							span_notice("You successfully break out of [src]!"))
+		bust_open()
+	else
+		if(user.loc == src) //so we don't get the message if we resisted multiple times and succeeded.
+			to_chat(user, span_warning("You fail to break out of [src]!"))
+
+
+/obj/structure/closet/body_bag/environmental/prisoner/bust_open()
+	SIGNAL_HANDLER
+	sinched = FALSE
+	// We don't break the bag, because the buckles were backed out as opposed to fully broken.
+	open()
+
+/obj/structure/closet/body_bag/environmental/prisoner/RightClick(mob/user, modifiers)
+	if(!user.canUseTopic(src, BE_CLOSE) || !isturf(loc))
+		return
+	if(!opened)
+		togglelock(user)
+	return TRUE
+
+/obj/structure/closet/body_bag/environmental/prisoner/togglelock(mob/living/user, silent)
```

Comment:
The observant among you may have noticed that there is no check to make sure you can't sinch/unsinch the bag from its interior. This is for debugging purposes, and will be corrected later.